libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/util/config/ConfigManager.java

```diff
@@ -0,0 +1,472 @@
+     * This will only return values from the config-sources: not values from {@link System#getProperty}.
+     */
+    public Map<String, String> getAllConfigValues() {
+        HashMap<String, String> result = new HashMap<>(previousConfigs.size());
+        previousConfigs.forEach((configKey, configEntry) -> result.put(configKey, translateConfigValue(configKey, configEntry.value)));
+        return result;
+    }
+
+    /** Stop reloading config source and update config-values  */
+    @Override
+    public void close() {
+        isClosed = true;
+        while ((reloadSourcesThread != null) && reloadSourcesThread.isAlive()) {
+            reloadSourcesThread.interrupt();
+            try {
+                Thread.sleep(10);
+            } catch (InterruptedException ignored) {
+            }
+        }
+    }
+
+    /** Overridable: like {@link System#getProperty(String)} */
+    protected String systemGetProperty(String key) {
+        return System.getProperty(key);
+    }
+
+    /** Overridable: like {@link System#clearProperty(String)} */
+    protected void systemClearProperty(String key) {
+        System.clearProperty(key);
+    }
+
+    /** Overridable: like {@link System#setProperty(String, String)} (property-value is NOT translated by {@link #translateConfigValue}) */
+    protected void systemSetProperty(String key, String value) {
+        System.setProperty(key, value);
+    }
+
+    /**
+     * Overridable: whenever getting a config-value (either if it originated from a config-source or from {@link #systemGetProperty})
+     *  pass the value through this method.
+     * This will affect the result of {@link #getConfigValue} and
+     *  {@link com.yahoo.athenz.common.server.util.config.dynamic.DynamicConfig#get()} and
+     *  {@link com.yahoo.athenz.common.server.util.config.dynamic.DynamicConfig.ChangeCallback#configChanged}
+     */
+    protected String translateConfigValue(@Nonnull String configKey, @Nullable String configValue) {
+        return configValue;
+    }
+
+    /** Start a thread to periodically reload configs */
+    private Thread periodicallyReloadConfigs(String reloadConfigKey, long reloadDefaultValue, TimeUnit reloadTimeUnit) {
+        DynamicConfigDuration interval = new DynamicConfigDuration(this, reloadConfigKey, reloadDefaultValue, reloadTimeUnit);
+        Thread reloadSourcesThread = new Thread(() -> {
+            while (!isClosed) {
+                try {
+                    interval.sleep(sleepMilliseconds -> Math.max(sleepMilliseconds, MINIMAL_PERIODIC_RELOAD_INTERVAL));
+                } catch (InterruptedException ignored) {
+                }
+                reloadAllConfigs();
+            }
+        });
+        reloadSourcesThread.setName("ConfigManagerReloader");
+        reloadSourcesThread.setDaemon(true);
+        reloadSourcesThread.start();
+        return reloadSourcesThread;
+    }
+
+    /** Given a list of changes - update System-properties, and call change-callbacks */
```

Comment:
the description seems misleading since the the method does not update system properties. it only calls the change callbacks. I also noticed this reference to updating system properties in the other methods where digestChangeLogs was being called so those need to be fixed as well.

Reply:
Fixed